// @lightprotocol/compressed-token/dist/cjs/browser/index.cjs
'use strict';
var e = require('./types-pEpHwEnq.cjs');
(require('@solana/web3.js'), require('@lightprotocol/stateless.js'));
var t = require('@solana/spl-token');
(require('@coral-xyz/borsh'),
    (exports.ADD_TOKEN_POOL_DISCRIMINATOR = e.ADD_TOKEN_POOL_DISCRIMINATOR),
    (exports.APPROVE_DISCRIMINATOR = e.APPROVE_DISCRIMINATOR),
    Object.defineProperty(exports, 'Action', {
        enumerable: 1,
        get: function () {
            return e.Action;
        },
    }),
    (exports.BASE_RENT_PER_EPOCH = e.BASE_RENT_PER_EPOCH),
    (exports.BATCH_COMPRESS_DISCRIMINATOR = e.BATCH_COMPRESS_DISCRIMINATOR),
    (exports.COMPRESSIBLE_CTOKEN_ACCOUNT_SIZE =
        e.COMPRESSIBLE_CTOKEN_ACCOUNT_SIZE),
    (exports.COMPRESSIBLE_CTOKEN_RENT_PER_EPOCH =
        e.COMPRESSIBLE_CTOKEN_RENT_PER_EPOCH),
    (exports.COMPRESSION_COST = e.COMPRESSION_COST),
    (exports.COMPRESSION_INCENTIVE = e.COMPRESSION_INCENTIVE),
    (exports.COMPRESS_SPL_TOKEN_ACCOUNT_DISCRIMINATOR =
        e.COMPRESS_SPL_TOKEN_ACCOUNT_DISCRIMINATOR),
    (exports.CPI_AUTHORITY_SEED = e.CPI_AUTHORITY_SEED),
    (exports.CREATE_TOKEN_POOL_DISCRIMINATOR =
        e.CREATE_TOKEN_POOL_DISCRIMINATOR),
    (exports.CompressedTokenInstructionDataApproveLayout =
        e.CompressedTokenInstructionDataApproveLayout),
    (exports.CompressedTokenInstructionDataRevokeLayout =
        e.CompressedTokenInstructionDataRevokeLayout),
    (exports.CompressedTokenInstructionDataTransferLayout =
        e.CompressedTokenInstructionDataTransferLayout),
    (exports.CompressedTokenProgram = e.CompressedTokenProgram),
    (exports.CpiContextLayout = e.CpiContextLayout),
    (exports.DECOMPRESS_ACCOUNTS_IDEMPOTENT_DISCRIMINATOR =
        e.DECOMPRESS_ACCOUNTS_IDEMPOTENT_DISCRIMINATOR),
    (exports.DEFAULT_COMPRESSIBLE_CONFIG = e.DEFAULT_COMPRESSIBLE_CONFIG),
    (exports.DEFAULT_PREPAY_EPOCHS = e.DEFAULT_PREPAY_EPOCHS),
    (exports.DEFAULT_WRITE_TOP_UP = e.DEFAULT_WRITE_TOP_UP),
    (exports.DelegatedTransferLayout = e.DelegatedTransferLayout),
    (exports.ERROR_MIXED_TREE_TYPES = e.ERROR_MIXED_TREE_TYPES),
    (exports.ERROR_NO_ACCOUNTS_FOUND = e.ERROR_NO_ACCOUNTS_FOUND),
    Object.defineProperty(exports, 'ExtensionType', {
        enumerable: 1,
        get: function () {
            return e.ExtensionType;
        },
    }),
    (exports.IDL = e.IDL),
    (exports.LIGHT_TOKEN_CONFIG = e.LIGHT_TOKEN_CONFIG),
    (exports.LIGHT_TOKEN_RENT_SPONSOR = e.LIGHT_TOKEN_RENT_SPONSOR),
    (exports.LightTokenProgram = e.CompressedTokenProgram),
    (exports.MAX_COMBINED_BATCH_BYTES = e.MAX_COMBINED_BATCH_BYTES),
    (exports.MAX_LOAD_ONLY_BATCH_BYTES = e.MAX_LOAD_ONLY_BATCH_BYTES),
    (exports.MAX_TOP_UP = e.MAX_TOP_UP),
    (exports.MAX_TRANSACTION_SIZE = e.MAX_TRANSACTION_SIZE),
    (exports.MINT_TO_DISCRIMINATOR = e.MINT_TO_DISCRIMINATOR),
    (exports.POOL_SEED = e.POOL_SEED),
    (exports.RENT_PER_BYTE_PER_EPOCH = e.RENT_PER_BYTE_PER_EPOCH),
    (exports.REVOKE_DISCRIMINATOR = e.REVOKE_DISCRIMINATOR),
    (exports.SLOTS_PER_RENT_EPOCH = e.SLOTS_PER_RENT_EPOCH),
    (exports.SPL_TOKEN_MINT_RENT_EXEMPT_BALANCE =
        e.SPL_TOKEN_MINT_RENT_EXEMPT_BALANCE),
    (exports.TOTAL_COMPRESSION_COST = e.TOTAL_COMPRESSION_COST),
    (exports.TRANSFER_DISCRIMINATOR = e.TRANSFER_DISCRIMINATOR),
    Object.defineProperty(exports, 'TokenDataVersion', {
        enumerable: 1,
        get: function () {
            return e.TokenDataVersion;
        },
    }),
    (exports.addSplInterfaces = e.addSplInterfaces),
    (exports.addTokenPoolAccountsLayout = e.addTokenPoolAccountsLayout),
    (exports.addTokenPools = e.addTokenPools),
    (exports.approve = e.approve),
    (exports.approveAccountsLayout = e.approveAccountsLayout),
    (exports.approveAndMintTo = e.approveAndMintTo),
    (exports.batchCompressLayout = e.batchCompressLayout),
    (exports.calculateCompressibleLoadComputeUnits =
        e.calculateCompressibleLoadComputeUnits),
    (exports.calculateFeePayerCostAtCreation =
        e.calculateFeePayerCostAtCreation),
    (exports.checkMint = e.checkMint),
    (exports.checkSplInterfaceInfo = e.checkSplInterfaceInfo),
    (exports.checkTokenPoolInfo = e.checkTokenPoolInfo),
    (exports.compress = e.compress),
    (exports.compressSplTokenAccount = e.compressSplTokenAccount),
    (exports.compressSplTokenAccountInstructionDataLayout =
        e.compressSplTokenAccountInstructionDataLayout),
    (exports.convertTokenDataToAccount = e.convertTokenDataToAccount),
    (exports.createAssociatedCTokenAccountIdempotentInstruction =
        e.createAssociatedCTokenAccountIdempotentInstruction),
    (exports.createAssociatedCTokenAccountInstruction =
        e.createAssociatedCTokenAccountInstruction),
    (exports.createAssociatedTokenAccountInterfaceIdempotentInstruction =
        e.createAssociatedTokenAccountInterfaceIdempotentInstruction),
    (exports.createAssociatedTokenAccountInterfaceInstruction =
        e.createAssociatedTokenAccountInterfaceInstruction),
    (exports.createAtaInterface = e.createAtaInterface),
    (exports.createAtaInterfaceIdempotent = e.createAtaInterfaceIdempotent),
    (exports.createAtaInterfaceIdempotentInstruction =
        e.createAtaInterfaceIdempotentInstruction),
    (exports.createDecompressInterfaceInstruction =
        e.createDecompressInterfaceInstruction),
    (exports.createDecompressOutputState = e.createDecompressOutputState),
    (exports.createLightTokenTransferInstruction =
        e.createLightTokenTransferInstruction),
    (exports.createLoadAccountsParams = e.createLoadAccountsParams),
    (exports.createLoadAtaInstructionsFromInterface =
        e.createLoadAtaInstructionsFromInterface),
    (exports.createMint = e.createMint),
    (exports.createMintInstruction = e.createMintInstruction),
    (exports.createMintInterface = e.createMintInterface),
    (exports.createMintToCompressedInstruction =
        e.createMintToCompressedInstruction),
    (exports.createMintToInstruction = e.createMintToInstruction),
    (exports.createMintToInterfaceInstruction =
        e.createMintToInterfaceInstruction),
    (exports.createRemoveMetadataKeyInstruction =
        e.createRemoveMetadataKeyInstruction),
    (exports.createSplInterface = e.createSplInterface),
    (exports.createTokenMetadata = e.createTokenMetadata),
    (exports.createTokenPool = e.createTokenPool),
    (exports.createTokenPoolAccountsLayout = e.createTokenPoolAccountsLayout),
    (exports.createTokenProgramLookupTable = e.createTokenProgramLookupTable),
    (exports.createTransferInterfaceInstructions =
        e.createTransferInterfaceInstructions),
    (exports.createTransferOutputState = e.createTransferOutputState),
    (exports.createUnwrapInstruction = e.createUnwrapInstruction),
    (exports.createUnwrapInstructions = e.createUnwrapInstructions),
    (exports.createUpdateFreezeAuthorityInstruction =
        e.createUpdateFreezeAuthorityInstruction),
    (exports.createUpdateMetadataAuthorityInstruction =
        e.createUpdateMetadataAuthorityInstruction),
    (exports.createUpdateMetadataFieldInstruction =
        e.createUpdateMetadataFieldInstruction),
    (exports.createUpdateMintAuthorityInstruction =
        e.createUpdateMintAuthorityInstruction),
    (exports.createWrapInstruction = e.createWrapInstruction),
    (exports.decodeApproveInstructionData = e.decodeApproveInstructionData),
    (exports.decodeBatchCompressInstructionData =
        e.decodeBatchCompressInstructionData),
    (exports.decodeCompressSplTokenAccountInstructionData =
        e.decodeCompressSplTokenAccountInstructionData),
    (exports.decodeMintToInstructionData = e.decodeMintToInstructionData),
    (exports.decodeRevokeInstructionData = e.decodeRevokeInstructionData),
    (exports.decodeTokenMetadata = e.decodeTokenMetadata),
    (exports.decodeTransferInstructionData = e.decodeTransferInstructionData),
    (exports.decompress = e.decompress),
    (exports.decompressDelegated = e.decompressDelegated),
    (exports.decompressInterface = e.decompressInterface),
    (exports.deriveSplInterfaceInfo = e.deriveSplInterfaceInfo),
    (exports.deriveTokenPoolInfo = e.deriveTokenPoolInfo),
    (exports.deserializeMint = e.deserializeMint),
    (exports.encodeApproveInstructionData = e.encodeApproveInstructionData),
    (exports.encodeBatchCompressInstructionData =
        e.encodeBatchCompressInstructionData),
    (exports.encodeCompressSplTokenAccountInstructionData =
        e.encodeCompressSplTokenAccountInstructionData),
    (exports.encodeMintToInstructionData = e.encodeMintToInstructionData),
    (exports.encodeRevokeInstructionData = e.encodeRevokeInstructionData),
    (exports.encodeTokenMetadata = e.encodeTokenMetadata),
    (exports.encodeTransferInstructionData = e.encodeTransferInstructionData),
    (exports.estimateTransactionSize = e.estimateTransactionSize),
    (exports.extractTokenMetadata = e.extractTokenMetadata),
    (exports.freezeAccountsLayout = e.freezeAccountsLayout),
    (exports.getAccountInterface = e.getAccountInterface),
    (exports.getAssociatedTokenAddressInterface =
        e.getAssociatedTokenAddressInterface),
    (exports.getMintInterface = e.getMintInterface),
    (exports.getOrCreateAtaInterface = e.getOrCreateAtaInterface),
    (exports.getSplInterfaceInfos = e.getSplInterfaceInfos),
    (exports.getTokenPoolInfos = e.getTokenPoolInfos),
    (exports.groupAccountsByTreeType = e.groupAccountsByTreeType),
    (exports.isSingleSplInterfaceInfo = e.isSingleSplInterfaceInfo),
    (exports.isSingleTokenPoolInfo = e.isSingleTokenPoolInfo),
    (exports.mergeTokenAccounts = e.mergeTokenAccounts),
    (exports.mintTo = e.mintTo$1),
    (exports.mintToAccountsLayout = e.mintToAccountsLayout),
    (exports.mintToCToken = e.mintTo),
    (exports.mintToCompressed = e.mintToCompressed),
    (exports.mintToInterface = e.mintToInterface),
    (exports.mintToLayout = e.mintToLayout),
    (exports.packCompressedTokenAccounts = e.packCompressedTokenAccounts),
    (exports.parseCTokenCold = e.parseCTokenCold),
    (exports.parseCTokenHot = e.parseCTokenHot),
    (exports.parseMaybeDelegatedTransfer = e.parseMaybeDelegatedTransfer),
    (exports.parseTokenData = e.parseTokenData),
    (exports.removeMetadataKey = e.removeMetadataKey),
    (exports.rentPerEpoch = e.rentPerEpoch),
    (exports.revoke = e.revoke),
    (exports.revokeAccountsLayout = e.revokeAccountsLayout),
    (exports.selectAccountsByPreferredTreeType =
        e.selectAccountsByPreferredTreeType),
    (exports.selectInputsForAmount = e.selectInputsForAmount),
    (exports.selectMinCompressedTokenAccountsForDecompression =
        e.selectMinCompressedTokenAccountsForDecompression),
    (exports.selectMinCompressedTokenAccountsForTransfer =
        e.selectMinCompressedTokenAccountsForTransfer),
    (exports.selectMinCompressedTokenAccountsForTransferOrPartial =
        e.selectMinCompressedTokenAccountsForTransferOrPartial),
    (exports.selectSmartCompressedTokenAccountsForTransfer =
        e.selectSmartCompressedTokenAccountsForTransfer),
    (exports.selectSmartCompressedTokenAccountsForTransferOrPartial =
        e.selectSmartCompressedTokenAccountsForTransferOrPartial),
    (exports.selectSplInterfaceInfo = e.selectSplInterfaceInfo),
    (exports.selectSplInterfaceInfosForDecompression =
        e.selectSplInterfaceInfosForDecompression),
    (exports.selectTokenAccountsForApprove = e.selectTokenAccountsForApprove),
    (exports.selectTokenPoolInfo = e.selectTokenPoolInfo),
    (exports.selectTokenPoolInfosForDecompression =
        e.selectTokenPoolInfosForDecompression),
    (exports.serializeMint = e.serializeMint),
    (exports.sliceLast = e.sliceLast),
    (exports.sumUpTokenAmount = e.sumUpTokenAmount),
    (exports.thawAccountsLayout = e.thawAccountsLayout),
    (exports.toAccountInfo = e.toAccountInfo),
    (exports.toOffChainMetadataJson = e.toOffChainMetadataJson),
    (exports.toTokenPoolInfo = e.toTokenPoolInfo),
    (exports.transfer = e.transfer),
    (exports.transferAccountsLayout = e.transferAccountsLayout),
    (exports.transferDelegated = e.transferDelegated),
    (exports.transferInterface = e.transferInterface),
    (exports.unpackMintData = e.unpackMintData),
    (exports.unpackMintInterface = e.unpackMintInterface),
    (exports.updateFreezeAuthority = e.updateFreezeAuthority),
    (exports.updateMetadataAuthority = e.updateMetadataAuthority),
    (exports.updateMetadataField = e.updateMetadataField),
    (exports.updateMintAuthority = e.updateMintAuthority),
    (exports.validateSameTokenOwner = e.validateSameTokenOwner),
    (exports.wrap = e.wrap),
    Object.defineProperty(exports, 'AccountState', {
        enumerable: 1,
        get: function () {
            return t.AccountState;
        },
    }),
    (exports.createLoadAtaInstructions = async function (t, o, r, n, s, a) {
        return e.createLoadAtaInstructions(t, o, r, n, s, a, 0);
    }),
    (exports.getAtaInterface = async function (t, o, r, n, s, a) {
        return e.getAtaInterface(t, o, r, n, s, a);
    }),
    (exports.loadAta = async function (t, o, r, n, s, a, c) {
        return e.loadAta(t, o, r, n, s, a, c, 0);
    }));
