// @lightprotocol/compressed-token/dist/es/browser/unified/index.js
import { ComputeBudgetProgram as e } from '@solana/web3.js';
import {
    LIGHT_TOKEN_PROGRAM_ID as s,
    buildAndSignTx as o,
    sendAndConfirmTx as c,
} from '@lightprotocol/stateless.js';
import {
    g as l,
    J as f,
    c as M,
    l as N,
    o as L,
    L as h,
    bS as g,
    bT as K,
    N as B,
    B as J,
    bU as ea,
} from '../types-BSM3Bq3L.js';
export {
    b2 as ADD_TOKEN_POOL_DISCRIMINATOR,
    b0 as APPROVE_DISCRIMINATOR,
    ax as Action,
    b5 as BASE_RENT_PER_EPOCH,
    aZ as BATCH_COMPRESS_DISCRIMINATOR,
    bW as COMPRESSED_MINT_SEED,
    bb as COMPRESSIBLE_CTOKEN_ACCOUNT_SIZE,
    bd as COMPRESSIBLE_CTOKEN_RENT_PER_EPOCH,
    b8 as COMPRESSION_COST,
    b9 as COMPRESSION_INCENTIVE,
    a$ as COMPRESS_SPL_TOKEN_ACCOUNT_DISCRIMINATOR,
    aV as CPI_AUTHORITY_SEED,
    aX as CREATE_TOKEN_POOL_DISCRIMINATOR,
    bE as CompressedTokenInstructionDataApproveLayout,
    bF as CompressedTokenInstructionDataRevokeLayout,
    bk as CompressedTokenInstructionDataTransferLayout,
    C as CompressedTokenProgram,
    bj as CpiContextLayout,
    b3 as DECOMPRESS_ACCOUNTS_IDEMPOTENT_DISCRIMINATOR,
    F as DEFAULT_COMPRESSIBLE_CONFIG,
    be as DEFAULT_PREPAY_EPOCHS,
    bf as DEFAULT_WRITE_TOP_UP,
    bi as DelegatedTransferLayout,
    aG as ERROR_MIXED_TREE_TYPES,
    aF as ERROR_NO_ACCOUNTS_FOUND,
    aa as ExtensionType,
    bh as IDL,
    aR as LIGHT_TOKEN_CONFIG,
    aS as LIGHT_TOKEN_RENT_SPONSOR,
    b4 as MAX_TOP_UP,
    aY as MINT_TO_DISCRIMINATOR,
    aU as POOL_SEED,
    b6 as RENT_PER_BYTE_PER_EPOCH,
    b1 as REVOKE_DISCRIMINATOR,
    b7 as SLOTS_PER_RENT_EPOCH,
    aW as SPL_TOKEN_MINT_RENT_EXEMPT_BALANCE,
    ba as TOTAL_COMPRESSION_COST,
    a_ as TRANSFER_DISCRIMINATOR,
    aT as TokenDataVersion,
    aj as addSplInterfaces,
    bx as addTokenPoolAccountsLayout,
    ak as addTokenPools,
    ac as approve,
    bA as approveAccountsLayout,
    ad as approveAndMintTo,
    bm as batchCompressLayout,
    d as calculateCompressibleLoadComputeUnits,
    bg as calculateFeePayerCostAtCreation,
    aQ as checkMint,
    av as checkSplInterfaceInfo,
    aB as checkTokenPoolInfo,
    ae as compress,
    af as compressSplTokenAccount,
    bn as compressSplTokenAccountInstructionDataLayout,
    a4 as convertTokenDataToAccount,
    m as createAssociatedCTokenAccountIdempotentInstruction,
    k as createAssociatedCTokenAccountInstruction,
    n as createAssociatedTokenAccountInterfaceInstruction,
    H as createAtaInterface,
    I as createAtaInterfaceIdempotent,
    p as createAtaInterfaceIdempotentInstruction,
    D as createDecompressInterfaceInstruction,
    bP as createDecompressOutputState,
    E as createLightTokenTransferInstruction,
    a as createLoadAccountsParams,
    b as createLoadAtaInstructionsFromInterface,
    ag as createMint,
    i as createMintInstruction,
    G as createMintInterface,
    r as createMintToCompressedInstruction,
    q as createMintToInstruction,
    t as createMintToInterfaceInstruction,
    y as createRemoveMetadataKeyInstruction,
    ah as createSplInterface,
    j as createTokenMetadata,
    ai as createTokenPool,
    bw as createTokenPoolAccountsLayout,
    al as createTokenProgramLookupTable,
    bO as createTransferOutputState,
    A as createUnwrapInstruction,
    v as createUpdateFreezeAuthorityInstruction,
    x as createUpdateMetadataAuthorityInstruction,
    w as createUpdateMetadataFieldInstruction,
    u as createUpdateMintAuthorityInstruction,
    z as createWrapInstruction,
    bH as decodeApproveInstructionData,
    br as decodeBatchCompressInstructionData,
    bt as decodeCompressSplTokenAccountInstructionData,
    bp as decodeMintToInstructionData,
    bJ as decodeRevokeInstructionData,
    a7 as decodeTokenMetadata,
    bv as decodeTransferInstructionData,
    am as decompress,
    as as decompressDelegated,
    P as decompressInterface,
    bV as deriveCMintAddress,
    au as deriveSplInterfaceInfo,
    aA as deriveTokenPoolInfo,
    a5 as deserializeMint,
    bG as encodeApproveInstructionData,
    bq as encodeBatchCompressInstructionData,
    bs as encodeCompressSplTokenAccountInstructionData,
    bo as encodeMintToInstructionData,
    bI as encodeRevokeInstructionData,
    a8 as encodeTokenMetadata,
    bu as encodeTransferInstructionData,
    a9 as extractTokenMetadata,
    bX as findMintAddress,
    bC as freezeAccountsLayout,
    a0 as getAccountInterface,
    bZ as getAssociatedCTokenAddress,
    bY as getAssociatedCTokenAddressAndBump,
    Z as getMintInterface,
    aw as getSplInterfaceInfos,
    aC as getTokenPoolInfos,
    aH as groupAccountsByTreeType,
    bQ as isSingleSplInterfaceInfo,
    bR as isSingleTokenPoolInfo,
    an as mergeTokenAccounts,
    ao as mintTo,
    by as mintToAccountsLayout,
    R as mintToCToken,
    S as mintToCompressed,
    T as mintToInterface,
    bl as mintToLayout,
    aP as packCompressedTokenAccounts,
    a2 as parseCTokenCold,
    a1 as parseCTokenHot,
    bN as parseMaybeDelegatedTransfer,
    bM as parseTokenData,
    Y as removeMetadataKey,
    bc as rentPerEpoch,
    ap as revoke,
    bB as revokeAccountsLayout,
    aI as selectAccountsByPreferredTreeType,
    aK as selectMinCompressedTokenAccountsForDecompression,
    aL as selectMinCompressedTokenAccountsForTransfer,
    aM as selectMinCompressedTokenAccountsForTransferOrPartial,
    aN as selectSmartCompressedTokenAccountsForTransfer,
    aO as selectSmartCompressedTokenAccountsForTransferOrPartial,
    ay as selectSplInterfaceInfo,
    az as selectSplInterfaceInfosForDecompression,
    aJ as selectTokenAccountsForApprove,
    aD as selectTokenPoolInfo,
    aE as selectTokenPoolInfosForDecompression,
    a6 as serializeMint,
    O as sliceLast,
    bK as sumUpTokenAmount,
    bD as thawAccountsLayout,
    a3 as toAccountInfo,
    ab as toOffChainMetadataJson,
    at as toTokenPoolInfo,
    aq as transfer,
    bz as transferAccountsLayout,
    ar as transferDelegated,
    $ as unpackMintData,
    _ as unpackMintInterface,
    V as updateFreezeAuthority,
    X as updateMetadataAuthority,
    W as updateMetadataField,
    U as updateMintAuthority,
    bL as validateSameTokenOwner,
    Q as wrap,
} from '../types-BSM3Bq3L.js';
export { AccountState } from '@solana/spl-token';
import '@coral-xyz/borsh';
async function ta(a, e, t, s, o, n) {
    return l(a, e, t, s, o, n, 1);
}
function sa(a, e, t = 0, o = s, n) {
    if (!o.equals(s))
        throw new Error(
            'Please derive the unified ATA from the c-token program; balances across SPL, T22, and c-token are unified under the canonical c-token ATA.',
        );
    return f(a, e, t, o, n);
}
async function oa(a, e, t, s, o, n) {
    return M(a, e, t, s, o, n, 1);
}
async function na(a, t, n, r, i, u, T) {
    i ??= n;
    const I = await N(a, t, n, r, i, u, T, 1);
    if (null === I && !(await a.getAccountInfo(t))) {
        const T = L(i.publicKey, t, n.publicKey, r, s),
            { blockhash: I } = await a.getLatestBlockhash(),
            p = o(
                [e.setComputeUnitLimit({ units: 3e4 }), T],
                i,
                I,
                i.publicKey.equals(n.publicKey) ? [] : [n],
            );
        return c(a, p, u);
    }
    return I;
}
async function ra(a, e, t, s, o, n, r, c, i) {
    return h(a, e, t, s, o, n, r, void 0, c, i, 1);
}
async function ca(a, e, t, o, n = 0, r, c) {
    return g(a, e, t, o, n, r, c, s, K(s), 1);
}
async function ia(a, e, t, s, o, n, r) {
    return B(a, e, t, s, o, n, { ...r, wrap: 1 });
}
async function ua(a, e, t, s, o, n, r, c) {
    return J(a, e, t, s, o, n, r, void 0, c, 1);
}
async function Ta(a, e, t, s, o, n, r, c) {
    return ea(a, e, t, s, o, n, r, void 0, c);
}
export {
    L as createAssociatedTokenAccountInterfaceIdempotentInstruction,
    oa as createLoadAtaInstructions,
    ia as createTransferInterfaceInstructions,
    ua as createUnwrapInstructions,
    sa as getAssociatedTokenAddressInterface,
    ta as getAtaInterface,
    ca as getOrCreateAtaInterface,
    na as loadAta,
    ra as transferInterface,
    Ta as unwrap,
};
