// @lightprotocol/compressed-token/dist/cjs/browser/unified/index.cjs
'use strict';
var e = require('@solana/web3.js'),
    t = require('@lightprotocol/stateless.js'),
    o = require('../types-pEpHwEnq.cjs'),
    r = require('@solana/spl-token');
(require('@coral-xyz/borsh'),
    (exports.ADD_TOKEN_POOL_DISCRIMINATOR = o.ADD_TOKEN_POOL_DISCRIMINATOR),
    (exports.APPROVE_DISCRIMINATOR = o.APPROVE_DISCRIMINATOR),
    Object.defineProperty(exports, 'Action', {
        enumerable: 1,
        get: function () {
            return o.Action;
        },
    }),
    (exports.BASE_RENT_PER_EPOCH = o.BASE_RENT_PER_EPOCH),
    (exports.BATCH_COMPRESS_DISCRIMINATOR = o.BATCH_COMPRESS_DISCRIMINATOR),
    (exports.COMPRESSED_MINT_SEED = o.COMPRESSED_MINT_SEED),
    (exports.COMPRESSIBLE_CTOKEN_ACCOUNT_SIZE =
        o.COMPRESSIBLE_CTOKEN_ACCOUNT_SIZE),
    (exports.COMPRESSIBLE_CTOKEN_RENT_PER_EPOCH =
        o.COMPRESSIBLE_CTOKEN_RENT_PER_EPOCH),
    (exports.COMPRESSION_COST = o.COMPRESSION_COST),
    (exports.COMPRESSION_INCENTIVE = o.COMPRESSION_INCENTIVE),
    (exports.COMPRESS_SPL_TOKEN_ACCOUNT_DISCRIMINATOR =
        o.COMPRESS_SPL_TOKEN_ACCOUNT_DISCRIMINATOR),
    (exports.CPI_AUTHORITY_SEED = o.CPI_AUTHORITY_SEED),
    (exports.CREATE_TOKEN_POOL_DISCRIMINATOR =
        o.CREATE_TOKEN_POOL_DISCRIMINATOR),
    (exports.CompressedTokenInstructionDataApproveLayout =
        o.CompressedTokenInstructionDataApproveLayout),
    (exports.CompressedTokenInstructionDataRevokeLayout =
        o.CompressedTokenInstructionDataRevokeLayout),
    (exports.CompressedTokenInstructionDataTransferLayout =
        o.CompressedTokenInstructionDataTransferLayout),
    (exports.CompressedTokenProgram = o.CompressedTokenProgram),
    (exports.CpiContextLayout = o.CpiContextLayout),
    (exports.DECOMPRESS_ACCOUNTS_IDEMPOTENT_DISCRIMINATOR =
        o.DECOMPRESS_ACCOUNTS_IDEMPOTENT_DISCRIMINATOR),
    (exports.DEFAULT_COMPRESSIBLE_CONFIG = o.DEFAULT_COMPRESSIBLE_CONFIG),
    (exports.DEFAULT_PREPAY_EPOCHS = o.DEFAULT_PREPAY_EPOCHS),
    (exports.DEFAULT_WRITE_TOP_UP = o.DEFAULT_WRITE_TOP_UP),
    (exports.DelegatedTransferLayout = o.DelegatedTransferLayout),
    (exports.ERROR_MIXED_TREE_TYPES = o.ERROR_MIXED_TREE_TYPES),
    (exports.ERROR_NO_ACCOUNTS_FOUND = o.ERROR_NO_ACCOUNTS_FOUND),
    Object.defineProperty(exports, 'ExtensionType', {
        enumerable: 1,
        get: function () {
            return o.ExtensionType;
        },
    }),
    (exports.IDL = o.IDL),
    (exports.LIGHT_TOKEN_CONFIG = o.LIGHT_TOKEN_CONFIG),
    (exports.LIGHT_TOKEN_RENT_SPONSOR = o.LIGHT_TOKEN_RENT_SPONSOR),
    (exports.MAX_TOP_UP = o.MAX_TOP_UP),
    (exports.MINT_TO_DISCRIMINATOR = o.MINT_TO_DISCRIMINATOR),
    (exports.POOL_SEED = o.POOL_SEED),
    (exports.RENT_PER_BYTE_PER_EPOCH = o.RENT_PER_BYTE_PER_EPOCH),
    (exports.REVOKE_DISCRIMINATOR = o.REVOKE_DISCRIMINATOR),
    (exports.SLOTS_PER_RENT_EPOCH = o.SLOTS_PER_RENT_EPOCH),
    (exports.SPL_TOKEN_MINT_RENT_EXEMPT_BALANCE =
        o.SPL_TOKEN_MINT_RENT_EXEMPT_BALANCE),
    (exports.TOTAL_COMPRESSION_COST = o.TOTAL_COMPRESSION_COST),
    (exports.TRANSFER_DISCRIMINATOR = o.TRANSFER_DISCRIMINATOR),
    Object.defineProperty(exports, 'TokenDataVersion', {
        enumerable: 1,
        get: function () {
            return o.TokenDataVersion;
        },
    }),
    (exports.addSplInterfaces = o.addSplInterfaces),
    (exports.addTokenPoolAccountsLayout = o.addTokenPoolAccountsLayout),
    (exports.addTokenPools = o.addTokenPools),
    (exports.approve = o.approve),
    (exports.approveAccountsLayout = o.approveAccountsLayout),
    (exports.approveAndMintTo = o.approveAndMintTo),
    (exports.batchCompressLayout = o.batchCompressLayout),
    (exports.calculateCompressibleLoadComputeUnits =
        o.calculateCompressibleLoadComputeUnits),
    (exports.calculateFeePayerCostAtCreation =
        o.calculateFeePayerCostAtCreation),
    (exports.checkMint = o.checkMint),
    (exports.checkSplInterfaceInfo = o.checkSplInterfaceInfo),
    (exports.checkTokenPoolInfo = o.checkTokenPoolInfo),
    (exports.compress = o.compress),
    (exports.compressSplTokenAccount = o.compressSplTokenAccount),
    (exports.compressSplTokenAccountInstructionDataLayout =
        o.compressSplTokenAccountInstructionDataLayout),
    (exports.convertTokenDataToAccount = o.convertTokenDataToAccount),
    (exports.createAssociatedCTokenAccountIdempotentInstruction =
        o.createAssociatedCTokenAccountIdempotentInstruction),
    (exports.createAssociatedCTokenAccountInstruction =
        o.createAssociatedCTokenAccountInstruction),
    (exports.createAssociatedTokenAccountInterfaceIdempotentInstruction =
        o.createAssociatedTokenAccountInterfaceIdempotentInstruction),
    (exports.createAssociatedTokenAccountInterfaceInstruction =
        o.createAssociatedTokenAccountInterfaceInstruction),
    (exports.createAtaInterface = o.createAtaInterface),
    (exports.createAtaInterfaceIdempotent = o.createAtaInterfaceIdempotent),
    (exports.createAtaInterfaceIdempotentInstruction =
        o.createAtaInterfaceIdempotentInstruction),
    (exports.createDecompressInterfaceInstruction =
        o.createDecompressInterfaceInstruction),
    (exports.createDecompressOutputState = o.createDecompressOutputState),
    (exports.createLightTokenTransferInstruction =
        o.createLightTokenTransferInstruction),
    (exports.createLoadAccountsParams = o.createLoadAccountsParams),
    (exports.createLoadAtaInstructionsFromInterface =
        o.createLoadAtaInstructionsFromInterface),
    (exports.createMint = o.createMint),
    (exports.createMintInstruction = o.createMintInstruction),
    (exports.createMintInterface = o.createMintInterface),
    (exports.createMintToCompressedInstruction =
        o.createMintToCompressedInstruction),
    (exports.createMintToInstruction = o.createMintToInstruction),
    (exports.createMintToInterfaceInstruction =
        o.createMintToInterfaceInstruction),
    (exports.createRemoveMetadataKeyInstruction =
        o.createRemoveMetadataKeyInstruction),
    (exports.createSplInterface = o.createSplInterface),
    (exports.createTokenMetadata = o.createTokenMetadata),
    (exports.createTokenPool = o.createTokenPool),
    (exports.createTokenPoolAccountsLayout = o.createTokenPoolAccountsLayout),
    (exports.createTokenProgramLookupTable = o.createTokenProgramLookupTable),
    (exports.createTransferOutputState = o.createTransferOutputState),
    (exports.createUnwrapInstruction = o.createUnwrapInstruction),
    (exports.createUpdateFreezeAuthorityInstruction =
        o.createUpdateFreezeAuthorityInstruction),
    (exports.createUpdateMetadataAuthorityInstruction =
        o.createUpdateMetadataAuthorityInstruction),
    (exports.createUpdateMetadataFieldInstruction =
        o.createUpdateMetadataFieldInstruction),
    (exports.createUpdateMintAuthorityInstruction =
        o.createUpdateMintAuthorityInstruction),
    (exports.createWrapInstruction = o.createWrapInstruction),
    (exports.decodeApproveInstructionData = o.decodeApproveInstructionData),
    (exports.decodeBatchCompressInstructionData =
        o.decodeBatchCompressInstructionData),
    (exports.decodeCompressSplTokenAccountInstructionData =
        o.decodeCompressSplTokenAccountInstructionData),
    (exports.decodeMintToInstructionData = o.decodeMintToInstructionData),
    (exports.decodeRevokeInstructionData = o.decodeRevokeInstructionData),
    (exports.decodeTokenMetadata = o.decodeTokenMetadata),
    (exports.decodeTransferInstructionData = o.decodeTransferInstructionData),
    (exports.decompress = o.decompress),
    (exports.decompressDelegated = o.decompressDelegated),
    (exports.decompressInterface = o.decompressInterface),
    (exports.deriveCMintAddress = o.deriveCMintAddress),
    (exports.deriveSplInterfaceInfo = o.deriveSplInterfaceInfo),
    (exports.deriveTokenPoolInfo = o.deriveTokenPoolInfo),
    (exports.deserializeMint = o.deserializeMint),
    (exports.encodeApproveInstructionData = o.encodeApproveInstructionData),
    (exports.encodeBatchCompressInstructionData =
        o.encodeBatchCompressInstructionData),
    (exports.encodeCompressSplTokenAccountInstructionData =
        o.encodeCompressSplTokenAccountInstructionData),
    (exports.encodeMintToInstructionData = o.encodeMintToInstructionData),
    (exports.encodeRevokeInstructionData = o.encodeRevokeInstructionData),
    (exports.encodeTokenMetadata = o.encodeTokenMetadata),
    (exports.encodeTransferInstructionData = o.encodeTransferInstructionData),
    (exports.extractTokenMetadata = o.extractTokenMetadata),
    (exports.findMintAddress = o.findMintAddress),
    (exports.freezeAccountsLayout = o.freezeAccountsLayout),
    (exports.getAccountInterface = o.getAccountInterface),
    (exports.getAssociatedCTokenAddress = o.getAssociatedCTokenAddress),
    (exports.getAssociatedCTokenAddressAndBump =
        o.getAssociatedCTokenAddressAndBump),
    (exports.getMintInterface = o.getMintInterface),
    (exports.getSplInterfaceInfos = o.getSplInterfaceInfos),
    (exports.getTokenPoolInfos = o.getTokenPoolInfos),
    (exports.groupAccountsByTreeType = o.groupAccountsByTreeType),
    (exports.isSingleSplInterfaceInfo = o.isSingleSplInterfaceInfo),
    (exports.isSingleTokenPoolInfo = o.isSingleTokenPoolInfo),
    (exports.mergeTokenAccounts = o.mergeTokenAccounts),
    (exports.mintTo = o.mintTo$1),
    (exports.mintToAccountsLayout = o.mintToAccountsLayout),
    (exports.mintToCToken = o.mintTo),
    (exports.mintToCompressed = o.mintToCompressed),
    (exports.mintToInterface = o.mintToInterface),
    (exports.mintToLayout = o.mintToLayout),
    (exports.packCompressedTokenAccounts = o.packCompressedTokenAccounts),
    (exports.parseCTokenCold = o.parseCTokenCold),
    (exports.parseCTokenHot = o.parseCTokenHot),
    (exports.parseMaybeDelegatedTransfer = o.parseMaybeDelegatedTransfer),
    (exports.parseTokenData = o.parseTokenData),
    (exports.removeMetadataKey = o.removeMetadataKey),
    (exports.rentPerEpoch = o.rentPerEpoch),
    (exports.revoke = o.revoke),
    (exports.revokeAccountsLayout = o.revokeAccountsLayout),
    (exports.selectAccountsByPreferredTreeType =
        o.selectAccountsByPreferredTreeType),
    (exports.selectMinCompressedTokenAccountsForDecompression =
        o.selectMinCompressedTokenAccountsForDecompression),
    (exports.selectMinCompressedTokenAccountsForTransfer =
        o.selectMinCompressedTokenAccountsForTransfer),
    (exports.selectMinCompressedTokenAccountsForTransferOrPartial =
        o.selectMinCompressedTokenAccountsForTransferOrPartial),
    (exports.selectSmartCompressedTokenAccountsForTransfer =
        o.selectSmartCompressedTokenAccountsForTransfer),
    (exports.selectSmartCompressedTokenAccountsForTransferOrPartial =
        o.selectSmartCompressedTokenAccountsForTransferOrPartial),
    (exports.selectSplInterfaceInfo = o.selectSplInterfaceInfo),
    (exports.selectSplInterfaceInfosForDecompression =
        o.selectSplInterfaceInfosForDecompression),
    (exports.selectTokenAccountsForApprove = o.selectTokenAccountsForApprove),
    (exports.selectTokenPoolInfo = o.selectTokenPoolInfo),
    (exports.selectTokenPoolInfosForDecompression =
        o.selectTokenPoolInfosForDecompression),
    (exports.serializeMint = o.serializeMint),
    (exports.sliceLast = o.sliceLast),
    (exports.sumUpTokenAmount = o.sumUpTokenAmount),
    (exports.thawAccountsLayout = o.thawAccountsLayout),
    (exports.toAccountInfo = o.toAccountInfo),
    (exports.toOffChainMetadataJson = o.toOffChainMetadataJson),
    (exports.toTokenPoolInfo = o.toTokenPoolInfo),
    (exports.transfer = o.transfer),
    (exports.transferAccountsLayout = o.transferAccountsLayout),
    (exports.transferDelegated = o.transferDelegated),
    (exports.unpackMintData = o.unpackMintData),
    (exports.unpackMintInterface = o.unpackMintInterface),
    (exports.updateFreezeAuthority = o.updateFreezeAuthority),
    (exports.updateMetadataAuthority = o.updateMetadataAuthority),
    (exports.updateMetadataField = o.updateMetadataField),
    (exports.updateMintAuthority = o.updateMintAuthority),
    (exports.validateSameTokenOwner = o.validateSameTokenOwner),
    (exports.wrap = o.wrap),
    Object.defineProperty(exports, 'AccountState', {
        enumerable: 1,
        get: function () {
            return r.AccountState;
        },
    }),
    (exports.createLoadAtaInstructions = async function (e, t, r, n, s, a) {
        return o.createLoadAtaInstructions(e, t, r, n, s, a, 1);
    }),
    (exports.createTransferInterfaceInstructions = async function (
        e,
        t,
        r,
        n,
        s,
        a,
        c,
    ) {
        return o.createTransferInterfaceInstructions(
            e,
            t,
            r,
            n,
            s,
            a,
            Object.assign(Object.assign({}, c), { wrap: 1 }),
        );
    }),
    (exports.createUnwrapInstructions = async function (
        e,
        t,
        r,
        n,
        s,
        a,
        c,
        p,
    ) {
        return o.createUnwrapInstructions(e, t, r, n, s, a, c, void 0, p, 1);
    }),
    (exports.getAssociatedTokenAddressInterface = function (
        e,
        r,
        n = 0,
        s = t.LIGHT_TOKEN_PROGRAM_ID,
        a,
    ) {
        if (!s.equals(t.LIGHT_TOKEN_PROGRAM_ID))
            throw new Error(
                'Please derive the unified ATA from the c-token program; balances across SPL, T22, and c-token are unified under the canonical c-token ATA.',
            );
        return o.getAssociatedTokenAddressInterface(e, r, n, s, a);
    }),
    (exports.getAtaInterface = async function (e, t, r, n, s, a) {
        return o.getAtaInterface(e, t, r, n, s, a, 1);
    }),
    (exports.getOrCreateAtaInterface = async function (
        e,
        r,
        n,
        s,
        a = 0,
        c,
        p,
    ) {
        return o._getOrCreateAtaInterface(
            e,
            r,
            n,
            s,
            a,
            c,
            p,
            t.LIGHT_TOKEN_PROGRAM_ID,
            o.getAtaProgramId(t.LIGHT_TOKEN_PROGRAM_ID),
            1,
        );
    }),
    (exports.loadAta = async function (r, n, s, a, c, p, T) {
        null != c || (c = s);
        const I = await o.loadAta(r, n, s, a, c, p, T, 1);
        if (null === I && !(await r.getAccountInfo(n))) {
            const T =
                    o.createAssociatedTokenAccountInterfaceIdempotentInstruction(
                        c.publicKey,
                        n,
                        s.publicKey,
                        a,
                        t.LIGHT_TOKEN_PROGRAM_ID,
                    ),
                { blockhash: I } = await r.getLatestBlockhash(),
                u = t.buildAndSignTx(
                    [
                        e.ComputeBudgetProgram.setComputeUnitLimit({
                            units: 3e4,
                        }),
                        T,
                    ],
                    c,
                    I,
                    c.publicKey.equals(s.publicKey) ? [] : [s],
                );
            return t.sendAndConfirmTx(r, u, p);
        }
        return I;
    }),
    (exports.transferInterface = async function (e, t, r, n, s, a, c, p, T) {
        return o.transferInterface(e, t, r, n, s, a, c, void 0, p, T, 1);
    }),
    (exports.unwrap = async function (e, t, r, n, s, a, c, p) {
        return o.unwrap(e, t, r, n, s, a, c, void 0, p);
    }));
